### PR TITLE
Adding entity node to core

### LINKED ISSUE #6627: [CT-1831] [Feature] Adding the `entity` node
Part of #6626 
----
### What this issue entails
- ✅ Add the `entity` node to dbt-core
- ✅ Add the `name`, `model`, `description` properties to the entity node
- ✅ Create the `entity` output in the manifest 
- ✅ Allow `entities` to be disabled

### PR DESCRIPTION
**Pull request by callum-mcdata**
Part of #6626 
resolves #6627 

### Description
Following the example in Gerda's initial introduction of metrics, this PR introduces the entity node.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by callum-mcdata** [comment]
Getting some strange errors on unit tests, such as the one highlighted below:
```
  ERROR collecting tests/unit/test_events.py _____________________________________________________
tests/unit/test_events.py:178: in <module>
    FinishedRunningStats(stat_line="", execution="", execution_time=0),
E   NameError: name 'FinishedRunningStats' is not defined
```

Not entirely sure what might be causing this issue

**Comment by callum-mcdata** [comment]
All right I fixed the issue above and I believe this PR is ready to go! Note that it is failing a number of artifact tests but that is to be expected - I didn't update them as this is just the first draft of what the entities node will look like. Not sure if we want to spend the time updating tests with each PR change or if we want to add all the changes in at the end.

File: core/dbt/graph/selector_methods.py [hunk 1]
```diff
@@ -118,6 +120,14 @@ def metric_nodes(self, included_nodes: Set[UniqueId]) -> Iterator[Tuple[UniqueId
                 continue
             yield unique_id, metric
 
+    def entity_nodes(self, included_nodes: Set[UniqueId]) -> Iterator[Tuple[UniqueId, Entity]]:
+
+        for key, metric in self.manifest.entities.items():
```
Comment (gshank) [inline core/dbt/graph/selector_methods.py:125]: Shouldn't this be "for key, entity" and not "for key, metric"?

File: testing-project/postgres/dbt_project.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Name your project! Project names should contain only lowercase characters
```
Comment (gshank) [inline testing-project/postgres/dbt_project.yml:1]: This testing-project will have to be removed before merging to main. Why is it checked in here?

**Review by gshank — APPROVED** [review]
Looks good!  There's so much duplicate boilerplate here that it makes me want to generalize it and make it more straightforward. Perhaps as part of the pluggable models effort...

**Pull request merged**